service/internal/builder/host_wrapper.go

```diff
@@ -38,3 +40,15 @@ func (hw *hostWrapper) ReportFatalError(err error) {
 	hw.Logger.Error("Component fatal error", zap.Error(err))
 	hw.Host.ReportFatalError(err)
 }
+
+// RegisterZPages is used by zpages extension to register handles from service.
+// When the wrapper is passed to the extension it won't be successful when casting
+// the interface, for the time being expose the interface here.
+// TODO: Find a better way to add the service zpages to the extension. This a temporary fix.
```

Comment:
@bogdandrutu right now my idea to avoid this service/zpagesextension is to expose the RegisterZPages on the zpagesextension package. This way objects that want to expose zpages can register their pages. The service could do the same. The downside is that the service will have a dependency on the zpagesextension. Any other ideas?